packages/talos_agent/src/agent/errors.rs

```diff
@@ -0,0 +1,55 @@
+use crate::agent::errors::AgentErrorKind::{CertificationTimout, Messaging};
+use crate::agent::model::CertifyRequestChannelMessage;
+use crate::messaging::errors::MessagingError;
+use std::error::Error;
+use strum::Display;
+use thiserror::Error as ThisError;
+use tokio::sync::mpsc::error::SendError;
+
+#[derive(Debug, Display)]
+pub enum AgentErrorKind {
+    Certification { xid: String },
+    CertificationTimout { xid: String, elapsed_ms: u128 },
+    Messaging,
+    Internal,
+}
+
+#[derive(Debug, ThisError)]
+#[error("Talos agent error: '{kind}'.\nReason: {reason}\nCause: {cause:?}")]
+pub struct AgentError {
+    pub kind: AgentErrorKind,
+    pub reason: String,
+    pub cause: Option<Box<dyn Error + Send>>,
```

Comment:
Not sure about the advantage of using `Box<dyn Error + Send>` in the cause vs just using `String` and mapping the internal error as `e.to_string()`. As all errors implement `Display` and usually `e.to_string()` will have sufficient information. 

Could you check the advantage of boxing and sending the full error here?

Also in rust library coding `Box<dyn Error>` is generally discouraged as the error is ambiguous to the consumer of the library

Reply:
I have changed it to be Optional<String>. But commit will be in the next branch where I am covering code with tests.